qiskit_experiments/calibration/calibrations.py

```diff
@@ -0,0 +1,438 @@
+    including the index of the channels. The parameter values are stored in a
+    dict where parameters are keys.
+    """
+
+    def __init__(self, backend: Backend):
+        """
+        Initialize the instructions from a given backend.
+
+        Args:
+            backend: The backend from which to get the configuration.
+        """
+
+        self._n_qubits = backend.configuration().num_qubits
+        self._n_uchannels = backend.configuration().n_uchannels
+        self._properties = backend.properties()
+        self._config = backend.configuration()
+        self._params = {}
+
+        # Required because copying a template schedule creates new paramters with new IDs.
+        self._parameter_map = {}
+        self._schedules = {}
+
+    def add_schedules(self, schedules: Union[Schedule, List[Schedule]]):
+        """
+        Add a schedule and register the parameters.
+
+        Args:
+            schedules: The schedule to add.
+
+        Raises:
+            CalibrationError: If the parameterized channel index is not formatted
+                following index1.index2...
+        """
+        if isinstance(schedules, Schedule):
+            schedules = [schedules]
+
+        for schedule in schedules:
+
+            # check that channels, if parameterized, have the proper name format.
+            # pylint: disable = raise-missing-from
+            for ch in schedule.channels:
+                if isinstance(ch.index, Parameter):
+                    try:
+                        [int(index) for index in ch.index.name.split(".")]
+                    except ValueError:
+                        raise CalibrationError(
+                            "Parameterized channel must have a name "
+                            "formatted following index1.index2..."
+                        )
+
+            self._schedules[schedule.name] = schedule
+
+            for param in schedule.parameters:
+                self._parameter_map[ParameterKey(schedule.name, param.name)] = param
+                if param not in self._params:
+                    self._params[param] = {}
+
+    @property
+    def parameters(self) -> Dict[Parameter, Set]:
+        """
+        Returns a dictionary of parameters managed by the calibrations definition. The value
+        of the dict is the schedule in which the parameter appears. Parameters that are not
+        attached to a schedule will have None as a key.
+        """
+        parameters = {}
+        for key, param in self._parameter_map.items():
+            schedule_name = key.schedule
+
+            if param not in parameters:
+                parameters[param] = {schedule_name}
+            else:
+                parameters[param].add(schedule_name)
+
+        return parameters
+
+    def add_parameter_value(
+        self,
+        value: ParameterValue,
+        param: Union[Parameter, str],
+        qubits: Tuple[int, ...],
+        schedule: Union[Schedule, str] = None,
```

Comment:
See PR #20. Also remember that many of the functions of `Calibrations` will be used by experiments rather than users. I don't expect that users will be directly calling functions like `add_parameter_value` in e.g. notebooks.